services/checks/checks.go

```diff
@@ -676,6 +672,34 @@ func (s *Service) findTargets(serviceType models.ServiceType, minPMMAgentVersion
 	return targets, nil
 }
 
+// pickPMMAgent selects the first pmm-agent with version >= minPMMAgentVersion.
+func (s *Service) pickPMMAgent(agents []*models.Agent, minPMMAgentVersion *version.Parsed) *models.Agent {
+	if len(agents) == 0 {
+		return nil
+	}
+
+	if minPMMAgentVersion == nil {
+		return agents[0]
+	}
+
+	for _, a := range agents {
+		v, err := version.Parse(pointer.GetString(a.Version))
+		if err != nil {
+			s.l.Warnf("Failed to parse pmm-agent version: %s.", err)
+
+			continue
+		}
+
+		if v.Less(minPMMAgentVersion) {
+			continue
+		}
+
+		return a
+	}
+
+	return nil
+}
```

Comment:
That change looks like a miss-merge

Reply:
yeah, just noticed that and trying to figure out how it happened